nwb_conversion_tools/nwbconverter.py

```diff
@@ -143,15 +143,19 @@ def run_conversion(
         assert (not save_to_file and nwbfile_path is None) or nwbfile is None, \
             "Either pass a nwbfile_path location with save_to_file=True, or a nwbfile object, but not both!"
 
+        # Validate metadata
+        if metadata is None:
+            metadata = self.get_metadata()
+        else:
+            self.validate_metadata(metadata=metadata)
```

Comment:
@luiztauffer The key assumption of course being that everything from self..get_metadata() is ensured to pass validation? No edge cases where it could fail?

I know we have built-in tests for this that check each individual data interface metadata, but a converter can take deep unions of that, and I just wonder if that could ever have the potential to give a post-unification metadata that isn't valid?

Reply:
>@luiztauffer The key assumption of course being that everything from self..get_metadata() is ensured to pass validation? 

That's correct.

> but a converter can take deep unions of that, and I just wonder if that could ever have the potential to give a post-unification metadata that isn't valid?

in this case I believe the best thing to do would be override the run_conversion method of the edgy case converter
```
def run_conversion(…)
	# code that reorganizes metadata and make it schema compatible
	metadata = ….
	super().run_conversion(metadata, args...)
```

Reply:
or actually, just override the `Converter.get_metadata()` would do it.

Reply:
Seems more complicated than I was thinking - I was simply proposing that we remove the else and always validate metadata, regardless of if it was passed or auto-generated.

```suggestion
            metadata = self.get_metadata()
        self.validate_metadata(metadata=metadata)
```

Reply:
of course, that makes sense